cabal-install/tests/IntegrationTests2/haddock-project/dependencies/a.cabal

```diff
@@ -0,0 +1,10 @@
+name: a
```

Comment:
This whole directory (`haddock-project`) looks out of place to me. I think it should be deeper in the directory structure. But I'm not familiar with `IntegrationTests2`: I usually use `cabal-testsuite/PackageTests` for integration tests -- I think, it's the latest rewrite of the integration test suite (cf. https://github.com/haskell/cabal/tree/master/cabal-testsuite#design-notes), which implies that extending the previous versions (e.g. `IntegrationTests2`) should, perhaps, be discouraged.

Reply:
I am not familiar with the integration test suite and I wasn't aware that `IntegrationTests2` is deprecated.  Are there some docs how to write the tests using `cabal-test`?

Reply:
I pushed a test using the new example.  The problem with it is that it passes even without the fix in this branch, maybe there's another way to add something to the store?

Reply:
`withRepo` is the only way to depend on a package in the tests, as far as I know. But I'm by no means an expert...

If the test doesn't come along, you could try to create a description for Manual QA (https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md?rgh-link-date=2023-04-25T20%3A36%3A28Z#qa-notes), i.e. explain how a human being can make sure that your changes achive the goal (and restating the goal would also help). But we all know that tests are a better deal.

Reply:
The same test now leaves in `cabal-testsuite` (part of this patch: `cabal-testsuite/PackageTests/HaddockProject`). Should this be removed? I don't see a point to run the same test twice. Or is there any substantial difference between the two tests?